nighthawk/README.md

```diff
@@ -0,0 +1,154 @@
+# Nighthawk
+
+*A benchmarking tool based on Envoy*
+
+## Current state
+
+The nighthawk client supports HTTP/1.1 and HTTP/2 over http and https. (NOTE: https certificates are not yet validated).
+
+## Prerequisites
+
+### Ubuntu
+
+1. Install the latest version of [Bazel](https://bazel.build/versions/master/docs/install.html) in your environment.
+
+2. Install external dependencies libtool, cmake, ninja, realpath and curl libraries separately.
+On Ubuntu, run the following command:
+
+``` bash
+
+# For Nighthawk / Envoy. 
+# Realpath may give an error, when it does that you probably already have it.
+
+sudo apt-get install \
+   libtool \
+   cmake \
+   realpath \
+   clang-format-7 \
+   automake \
+   ninja-build \
+   curl \
+   unzip
+
+# For tools/stats.py
+sudo apt-get install python3 python3-pip
+sudo pip3 install hdrhistogram jsonpickle
+```
+
+## Building and testing Nighthawk
+```zsh
+# TODO(oschaaf): Collect and list prerequisites. 
+# Currently mostly the same as Envoy.
+
+# build it
+bazel build //:nighthawk_client
+
+# test it
+bazel test //test:nighthawk_test
+```
+
+## Using the nighthawk client
+
+```
+➜ bazel-bin/nighthawk_client --help
+
+USAGE:
+
+   bazel-bin/nighthawk_client  [-v <trace|debug|info|warn|error>]
+                               [--concurrency <string>] [--h2] [--timeout
+                               <uint64_t>] [--duration <uint64_t>]
+                               [--connections <uint64_t>] [--rps
+                               <uint64_t>] [--] [--version] [-h] <uri
+                               format>
+
+
+Where:
+
+   -v <trace|debug|info|warn|error>,  --verbosity <trace|debug|info|warn
+      |error>
+     Verbosity of the output. Possible values: [trace, debug, info, warn,
+     error, critical]. The default level is 'info'.
+
+   --concurrency <string>
+     The number of concurrent event loops that should be used. Specify
+     'auto' to let nighthawk run leverage all (aligned) vCPUs. Note that
+     increasing this effectively multiplies configured --rps and
```

Comment:
increasing this results in an effective load multiplier combined with the configured `--rps` and `--connections` values.